scipy/linalg/tests/test_lapack.py

```diff
@@ -1645,3 +1645,158 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def test_gttrf_gttrs():
+    #%%
+    np.random.seed(42)
+    # use ?gttrf and ?gttrs to solve a random linear system for each dtype
+    for index, dtype in enumerate(DTYPES):
+        n = 10
+        # set appropriate test tolerance based on the data type
+        if index % 2 == 0:
+            rtol = 10**-(np.finfo(dtype).precision-1)
+        else:
+            rtol = 10**-(np.finfo(dtype).precision-3)
+        # create the matrix in accordance with the data type
+        if index < 2:
```

Comment:
let's replace the `if`-`else` logic by creating a function:
```
def my_rand(shape, dtype)
```
`shape` is a tuple. you'll use it to call `np.random.rand(*shape)`.
If `dtype` is one of the complex data types, you'll add a random imaginary component.
Before returning, you'll use `.astype(dtype=dtype)` to cast the array to the appropriate `dtype`.
It will be useful here and in future PRs.